app/src/main/java/com/greenfox/tribesoflagopusandroid/fragments/BuildingsFragment.java

```diff
@@ -144,18 +127,33 @@ public void onFailure(Call<Building> call, Throwable t) {
         return rootView;
     }
 
+    public void getBuildingsFromAPI() {
+        ((MainActivity)getActivity()).switchToLoadingView();
+        apiService.getBuildings(preferences.getString(USER_ACCESS_TOKEN, "")).enqueue(new Callback<BuildingsResponse>() {
+            @Override
+            public void onResponse(Call<BuildingsResponse> call, Response<BuildingsResponse> response) {
+                buildingsAdapter.addAll(response.body().getBuildings());
+                listView.setAdapter(buildingsAdapter);
+                ((MainActivity) getActivity()).sitchToContentView();
+            }
+
+            @Override
+            public void onFailure(Call<BuildingsResponse> call, Throwable t) {
+            }
+        });
+    }
+
+    @Override
+    public void refreshActiveFragment() {
```

Comment:
@georgezsiga this is the method where this fragment gets refreshed, it doesnt know if its the active, the activity knows that. from this perspective its just refreshing